pypuppetdb/api/__init__.py

```diff
@@ -793,6 +796,14 @@ def reports(self, **kwargs):
         """
         reports = self._query('reports', **kwargs)
         for report in reports:
+            try:
+                code_id = report['code_id']
+                catalog_uuid = report['catalog_uuid']
+                cached_catalog_status = report['cached_catalog_status']
+            except KeyError:
+                code_id = None
+                catalog_uuid = None
+                cached_catalog_status = None
```

Comment:
This might be better as the following
```
code_id = report.get('code_id')
catalog_uuid = report.get('catalog_uuid')
cached_catalog_status = report.get('cached_catalog_status')
```